tests/components/smartthings/test_init.py

```diff
@@ -61,17 +61,23 @@ async def test_unrecoverable_api_errors_create_new_flow(
     )
 
     # Assert setup returns false
-    result = await smartthings.async_setup_entry(hass, config_entry)
-    assert not result
+    assert not await smartthings.async_setup_entry(hass, config_entry)
```

Comment:
We should use the core interfaces in the tests.
```suggestion
    assert not await hass.config_entries.async_setup(config_entry.entry_id)
```